core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2541,6 +2541,16 @@ private[spark] object Utils extends Logging {
     master == "local" || master.startsWith("local[")
   }
 
+  /**
+   * Push based shuffle can only be enabled when external shuffle service is enabled.
+   * In the initial version, we cannot support pushed based shuffle and adaptive execution
+   * at the same time. Will improve this in a later version.
```

Comment:
If we have a JIRA for this, could you mention here as `IDed TODO` like `TODO(SPARK-XXX)`, please?

Reply:
This comment is outdated. The current version can already support AQE with push-based shuffle.